### PR TITLE
planner: refine warnings about can't push down to TiFlash.

### PR DESCRIPTION
**Pull request by LittleFall**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/27522

### What is changed and how it works?

1. add warnings about column type `bit`, `set`, `geometry`, `unspecified`.
2. remove warnings about `non-mpp`.

### Check List

Tests 

- [x] Unit test

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: expression/expression.go [hunk 1]
```diff
@@ -1218,12 +1218,8 @@ func canScalarFuncPushDown(scalarFunc *ScalarFunction, pc PbConverter, storeType
...
 			})
 		}
-		if pc.sc.InExplainStmt {
-			storageName := storeType.Name()
-			if storeType == kv.UnSpecified {
-				storageName = "storage layer"
-			}
-			pc.sc.AppendWarning(errors.New("Scalar function '" + scalarFunc.FuncName.L + "'(signature: " + scalarFunc.Function.PbCode().String() + ", return type: " + scalarFunc.RetType.CompactStr() + ") can not be pushed to " + storageName))
+		if storeType == kv.TiFlash && pc.sc.InExplainStmt {
+			pc.sc.AppendWarning(errors.New("Scalar function '" + scalarFunc.FuncName.L + "'(signature: " + scalarFunc.Function.PbCode().String() + ", return type: " + scalarFunc.RetType.CompactStr() + ") is not supported to push down to TiFlash now."))
```
Comment (windtalker) [inline expression/expression.go:1222]: Why don't log erros if the store type is not TiFlash?

File: planner/core/testdata/enforce_mpp_suite_in.json [hunk 2]
```diff
@@ -41,7 +41,8 @@
       "EXPLAIN SELECT count(b) from t where a=1; -- 7. agg func has virtual column",
       "EXPLAIN SELECT count(*) from t group by b; -- 8. group by virtual column",
       "EXPLAIN SELECT count(a) from t group by md5(a); -- 10. scalar func not supported",
-      "EXPLAIN SELECT count(a) from t where c=1; -- 11. type supported"
+      "EXPLAIN SELECT count(a) from t where c=1; -- 11. type not supported",
+      "EXPLAIN SELECT count(a) from t where d=1; -- 11. type not supported"
```
Comment (windtalker) [inline planner/core/testdata/enforce_mpp_suite_in.json:45]: 11. type not supported => 12. type not supported

Comment (LittleFall) [reply planner/core/testdata/enforce_mpp_suite_in.json:45]: both sql are `case 11`.

**Review by windtalker — APPROVED** [review]
LGTM

**Review by rebelice — APPROVED** [review]

**Comment by rebelice** [comment]
@LittleFall you must fix the test `TestShowWarningsForExprPushdown`

**Pull request merged**